components/kyma-environment-broker/internal/orchestration/manager/manager.go

```diff
@@ -216,13 +219,49 @@ func (m *orchestrationManager) resolveWindowTime(beginTime, endTime time.Time) (
 
 	// if time window has already passed we wait until next day
 	if start.Before(n) && end.Before(n) {
-		start = start.AddDate(0, 0, 1)
-		end = end.AddDate(0, 0, 1)
+		day := firstAvailableDay(n.Day(), convertSliceOfDaysToMap(availableDays))
+		start = start.AddDate(0, 0, int(math.Abs(float64(day - n.Day()))))
```

Comment:
The day difference logic mentioned in my previous comment should be used also here